aws_auditing_list.py

```diff
@@ -38,376 +44,678 @@ def default(self, o):
-        self.nELB = elbv2_cost_region
 
     def get_ebs_resources(self, regions):
-        with open("ebs_pricing_list.json","r") as ebs:
+        with open("ebs_pricing_list.json", "r") as ebs:
             vol_pricing = json.load(ebs)
         sts_response = self.sts_client.get_caller_identity()
-        user_account = sts_response['Account']
-        vol_cost_region = {}   
-        snap_vol_region = {}
-        snapshot_count_region= {}
+        user_account = sts_response["Account"]
 
         for region_name in regions:
             snap_vol = []
-            snap_vol_region[region_name] = {}
-            vol_cost_region[region_name] = {}
-            snapshot_count_region[region_name] = {}
-            count_list = {}
-            count = 0
+            self.snap_vol_id[region_name] = {}
+            self.volume_ebs[region_name] = {
+                "attached": {},
+                "unattached": {},
+            }
+            attached_list = {}
+            unattached_list = {}
+            self.snapshot_ebs[region_name] = {}
             snap_vol = []
-            orphaned_snap_vol = []
-            tv = 0
-            tvc = 0
-            snapshot_count = 0
-            orphaned_snapshot_count = 0
-            snapshot_count_list = {}
-            total_volume_cost = 0
-            attached_instances = 0
-            unattached_instances = 0
-            total_instances = 0
-            total_elb_cost = 0
-            total_elbv2_cost = 0
-            standard_count = 0
-            gp2_count = 0
-            io1_count = 0
-            sc1_count = 0
-            st1_count = 0
-            gp2_price = 0
-            io1_price = 0
-            sc1_price = 0
-            st1_price = 0
-            standard_price = 0
-            gp2_vol_size = 0
-            io1_vol_size = 0
-            sc1_vol_size = 0
-            st1_vol_size = 0
-            standard_vol_size = 0
-
-            conn = self.connect("ec2", region_name=region_name)
+            attached_vol_list = []
+            unattached_vol_list = []
+            conn = self.connect_service_region(
+                "ec2", region_name=region_name
+            )
 
             volumes = conn.describe_volumes()
-            vol_length = len(volumes["Volumes"])
-            snapshots = conn.describe_snapshots(Filters=[
-                {
-                    'Name' : 'owner-id',
-                    'Values' : [
-                    str(user_account),
-                    ]
-                }
-            ])
-            snapshot_length = len(snapshots["Snapshots"])
-            for vol in volumes['Volumes']:
+            snapshots = conn.describe_snapshots(
+                Filters=[
+                    {
+                        "Name": "owner-id",
+                        "Values": [str(user_account)],
+                    }
+                ]
+            )
+
+            for vol in volumes["Volumes"]:
+                if len(vol["Attachments"]):
+                    if (
```

Comment:
`if not vol["VolumeId"] in attached_vol_list:` I would even consider shortening the `attached_vol_list` to `attached_vol` - same for the ones below